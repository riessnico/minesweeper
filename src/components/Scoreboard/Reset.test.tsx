import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { Reset } from './Reset';

describe('Reset button check', () => {
  const ResetWithDummyHandlerOnReset: React.FC = () => (
    <Reset onReset={() => null} />
  );

  it('Reset renders correctly', () => {
    const onReset = jest.fn();

    const { asFragment } = render(<Reset onReset={onReset} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render elemens with default state', () => {
    render(<ResetWithDummyHandlerOnReset />);

    expect(screen.getByText('ğŸ˜Š')).toBeInTheDocument();
  });
  it('onReset handler should be called', () => {
    const onReset = jest.fn();

    render(<Reset onReset={onReset} />);

    fireEvent.click(screen.getByText('ğŸ˜Š'));

    expect(onReset).toBeCalled();
  });
  it('should change state when onMouseDown and MouseUp events happens', () => {
    render(<ResetWithDummyHandlerOnReset />);

    fireEvent.mouseDown(screen.getByText('ğŸ˜Š'));

    expect(screen.getByText('ğŸ˜®')).toBeInTheDocument();

    fireEvent.mouseUp(screen.getByText('ğŸ˜®'));

    expect(screen.getByText('ğŸ˜Š')).toBeInTheDocument();
  });
  it('should change state when onMouseDown and MouseLeave events happens', () => {
    render(<ResetWithDummyHandlerOnReset />);

    fireEvent.mouseDown(screen.getByText('ğŸ˜Š'));

    expect(screen.getByText('ğŸ˜®')).toBeInTheDocument();

    fireEvent.mouseLeave(screen.getByText('ğŸ˜®'));

    expect(screen.getByText('ğŸ˜Š')).toBeInTheDocument();
  });
});
