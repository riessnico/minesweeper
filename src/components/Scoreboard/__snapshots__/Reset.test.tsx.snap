// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset button check Reset renders correctly 1`] = `
<DocumentFragment>
  <button
    class="css-mt04f9"
  >
    ðŸ˜Š
  </button>
</DocumentFragment>
`;
